src/targets/__tests__/pubDev.test.ts

```diff
@@ -0,0 +1,302 @@
+import { promises as fsPromises } from 'fs';
+import { platform } from 'os';
+import simpleGit from 'simple-git';
+import { PubDevTarget, targetSecrets } from '../pubDev';
+import { spawnProcess } from '../../utils/system';
+import { NoneArtifactProvider } from '../../artifact_providers/none';
+
+jest.mock('../../utils/system');
+jest.mock('../../utils/files', () => ({
+  ...jest.requireActual('../../utils/files'),
+  withTempDir: async (cb: (dir: string) => Promise<void>) => cb(TMP_DIR),
+}));
+
+jest.mock('os', () => ({
+  ...jest.requireActual('os'),
+  platform: jest.fn(() => 'darwin'),
```

Comment:
If the code supports two platforms, why do tests only run in one of them? We can use matrix tests.

Reply:
With that logic we should perform all tests on all possible platorms 😅  this is only to confirm valid path getter function.